samples/GalleryApp/ObjectDetectorPage.xaml.cs

```diff
@@ -0,0 +1,494 @@
+                    }
+
+                    // Update displayed results
+                    m_bboxRenderer.Render(objectDetections);
+
+                    // Update the displayed performance text
+                    UIPerfTextBlock.Text = $"bind: {m_bindTime.ToString("F2")}ms, eval: {m_evalTime.ToString("F2")}ms";
+                }
+                catch (TaskCanceledException)
+                {
+                    // no-op: we expect this exception when we change media sources
+                    // and can safely ignore/continue
+                }
+                catch (Exception ex)
+                {
+                    NotifyUser($"Exception while rendering results: {ex.Message}");
+                }
+            });
+        }
+
+        /// <summary>
+        /// Configure an IFrameSource from a StorageFile or MediaCapture instance
+        /// </summary>
+        /// <param name="source"></param>
+        /// <returns></returns>
+        private async Task ConfigureFrameSourceAsync(object source)
+        {
+            await m_lock.WaitAsync();
+            {
+                // Reset bitmap rendering component
+                UIProcessedPreview.Source = null;
+                m_processedBitmapSource = new SoftwareBitmapSource();
+                UIProcessedPreview.Source = m_processedBitmapSource;
+
+                // Clean up previous frame source
+                if (m_frameSource != null)
+                {
+                    m_frameSource.FrameArrived -= FrameSource_FrameAvailable;
+                    var disposableFrameSource = m_frameSource as IDisposable;
+                    if (disposableFrameSource != null)
+                    {
+                        // Lock disposal based on frame source consumers
+                        disposableFrameSource.Dispose();
+                    }
+                }
+
+                // Create new frame source and rgister a callback if the source fails along the way
+                m_frameSource = await FrameSourceFactory.CreateFrameSourceAsync(source, (sender, message) =>
+                {
+                    NotifyUser(message);
+                });
+
+                // TODO: Workaround for a bug in ObjectDetectorBinding when binding consecutively VideoFrames with Direct3DSurface and SoftwareBitmap
+                m_binding = await m_skill.CreateSkillBindingAsync() as ObjectDetectorBinding;
+            }
+            m_lock.Release();
+
+            // If we obtained a valid frame source, start it
+            if (m_frameSource != null)
+            {
+                m_frameSource.FrameArrived += FrameSource_FrameAvailable;
+                await m_frameSource.StartAsync();
+            }
+        }
+
+        /// <summary>
+        /// Print a message to the UI
+        /// </summary>
+        /// <param name="message"></param>
+        private void NotifyUser(String message)
```

Comment:
>NotifyUser [](start = 21, length = 10)

suggestion for future: consider moving this to a common helper class accessible to all sub pages

Reply:
Task 22491399: Refactor codes - NotifyUser

---
In reply to: [299503059](https://github.com/microsoft/WindowsVisionSkillsPreview/pull/39#discussion_r299503059) [](ancestors = 299503059)